app/controllers/utils/CascadeUpsert.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.utils
+
+import javax.inject.Singleton
+import play.api.libs.json._
+import uk.gov.hmrc.http.cache.client.CacheMap
+
+@Singleton
+class CascadeUpsert {
+
+  val funcMap: Map[String, (JsValue, CacheMap) => CacheMap] =
```

Comment:
This funcMap looks a bit weird to me. It's only used in one place and returns an empty Map. I don't know whether it can be removed in anyway. I assume due to the return type there is probably a reason for this.